DevCycle/DVCClient.swift

```diff
@@ -186,42 +181,31 @@ public class DVCClient {
     }
 
     public func track(_ event: DVCEvent) {
-        self.eventQueue.append(event)
+        self.eventQueue.queue(event)
     }
 
-    public func flushEvents() {
-        var eventsToFlushQueue: [DVCEvent] = self.eventQueue
-        eventsToFlushQueue.append(
-            contentsOf: self.aggregateEventQueue.variableDefaulted.map { (_: String, defaultedEvent: DVCEvent) -> DVCEvent in
-                defaultedEvent
-            }
-        )
-        eventsToFlushQueue.append(
-            contentsOf: self.aggregateEventQueue.variableEvaluated.map { (_: String, evaluatedEvent: DVCEvent) -> DVCEvent in
-                evaluatedEvent
+    public func flushEvents(callback: FlushCompletedHandler? = nil) {
+        guard let user = self.user else {
+            Log.error("Flushing events failed, user not defined")
+            return
+        }
+        guard let service = self.service else {
+            Log.error("Service not setup correctly")
```

Comment:
What service is this talking about? also not sure our users would understand what this means.

Reply:
also should these errors be sent to the `callback`?

Reply:
i'll rename the error messages to be more verbose that it's a client setup issue

Reply:
Should the errors be sent to the callback?

Reply:
i'm not sure, it's not necessarily an error with the flush method

Reply:
all the other methods throw instead of calling the callback with an error

Reply:
well for ObjC we are catching the errors and returning them in the callback: 
<img width="826" alt="image" src="https://user-images.githubusercontent.com/1219069/153439382-af911a7e-5e19-4aa9-949d-dacf4b8bb2e1.png">

Reply:
so should we be throwing here then? and caching the errors for the callback on the ObjC side?